spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -598,6 +601,142 @@ def show_widget(self):
+        directories in the switcher. Only handle file mode, where
+        `mode` is empty string.
+        Parameters
+        ----------
+        mode: str
+            The selected mode (open files "", symbol "@" or line ":").
+        """
+        if mode != '':
+            return
+
+        paths = self._execute_fzf_subprocess()
+        if paths == []:
+            return
+        # the paths that are opened in the editor need to be excluded because
+        # they are shown already in the switcher in the "editor" section.
+        open_files = self.get_plugin()._get_open_filenames()
+        for file in open_files:
+            if file in paths:
+                paths.remove(file)
+
+        is_unsaved = [False] * len(paths)
+        short_paths = shorten_paths(paths, is_unsaved)
+        section = self.get_title()
+
+        for i, (path, short_path) in enumerate(zip(paths, short_paths)):
+            title = osp.basename(path)
+            icon = get_file_icon(path)
+            description = osp.dirname(path)
+            if len(path) > 75:
+                description = short_path
+            is_last_item = (i+1 == len(paths))
+
+            self.get_plugin()._switcher.add_item(
+                title=title,
+                description=description,
+                icon=icon,
+                section=section,
+                data=path,
+                last_item=is_last_item
+            )
+        self.get_plugin()._switcher.set_current_row(0)
+
+    def handle_switcher_selection(self, item, mode, search_text):
+        """
+        Handle user selecting item in switcher.
+        If the selected item is not in the section of the switcher that
+        corresponds to this plugin, then ignore it. Otherwise, switch to
+        selected item in notebook plugin and hide the switcher.
+        ----------
+        item: object
+            The current selected item from the switcher list (QStandardItem).
+        mode: str
+            The current selected mode (open files "", symbol "@" or line ":").
+        search_text: str
+            Cleaned search/filter text.
+        """
+
+        if item.get_section() != self.get_title():
+            return
+
+        # Open file in editor
+        filename = item.get_data()
+        switcher = self.get_plugin()._switcher
+        switcher.sig_open_file_requested.emit(filename)
+        switcher.hide()
+
+    def handle_switcher_results(self, search_text):
+        """
+        Handle user typing in switcher to filter result.
+        Load switcher results when a search text is typed for projects.
+        Parameters
+        ----------
+        text: str
+            The current search text in the switcher dialog box.
+        """
+        switcher = self.get_plugin()._switcher
+        paths = self._execute_fzf_subprocess(search_text)
+
+        for row in range(switcher.get_model().rowCount()):
+            item = switcher.get_model().item(row)
```

Comment:
Maybe the switcher signal that is connected to this method could pass a list with the data of the items being displayed to not have to directly use the switcher model here?

Reply:
Yes, that's a better idea!